src/main/java/tds/content/repositories/impl/S3ItemDataRepository.java

```diff
@@ -61,6 +65,27 @@ public String findOne(final String itemDataPath) throws IOException {
         }
     }
 
+    @Override
+    public InputStream findResource(final String resourcePath) throws IOException {
+        final String resourceLocation = s3Properties.getItemPrefix() + buildPath(resourcePath);
+
+        try {
+            final S3Object item = s3Client.getObject(new GetObjectRequest(
+                s3Properties.getBucketName(), resourceLocation));
+
+            if (item == null) {
+                throw new IOException("Could not find file for " + resourceLocation);
+            }
+
+            return item.getObjectContent();
+        } catch (final AmazonS3Exception ex) {
+            if (ex.getStatusCode() == HttpStatus.SC_NOT_FOUND) {
+                throw new NotFoundException(String.format("Could not find resource at resource location %s", resourcePath));
```

Comment:
I believe S3 will throw a generic exception that results in a 5xx as opposed to a 4xx, hence why I'm re-throwing with an exception thats intercepted by our advice classes